app/org/sagebionetworks/bridge/play/controllers/ExternalIdControllerV4.java

```diff
@@ -0,0 +1,60 @@
+package org.sagebionetworks.bridge.play.controllers;
+
+import static org.sagebionetworks.bridge.Roles.DEVELOPER;
+import static org.sagebionetworks.bridge.Roles.RESEARCHER;
+
+import org.sagebionetworks.bridge.BridgeConstants;
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.accounts.ExternalIdentifier;
+import org.sagebionetworks.bridge.models.accounts.ExternalIdentifierInfo;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.services.ExternalIdService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import play.mvc.Result;
+
+@Controller("externalIdControllerV4")
+public class ExternalIdControllerV4 extends BaseController {
+
+    private ExternalIdService externalIdService;
+    
+    @Autowired
+    final void setExternalIdService(ExternalIdService externalIdService) {
+        this.externalIdService = externalIdService;
+    }
+    
+    public Result getExternalIdentifiers(String offsetKey, String pageSizeString, String idFilter,
+            String assignmentFilterString) {
+        getAuthenticatedSession(DEVELOPER, RESEARCHER);
+
+        Integer pageSize = BridgeUtils.getIntOrDefault(pageSizeString, BridgeConstants.API_DEFAULT_PAGE_SIZE);
+        Boolean assignmentFilter = (assignmentFilterString == null) ? null : "true".equals(assignmentFilterString);
+        
+        ForwardCursorPagedResourceList<ExternalIdentifierInfo> page = externalIdService.getExternalIds(
+                offsetKey, pageSize, idFilter, assignmentFilter);
+        return okResult(page);
+    }
```

Comment:
Because we version all the verbs at a particular URL, if we change the semantics of one verb (in this case, POST), I think we need to provide all the other verbs at that new URL too. Otherwise it will get quite confusing if you GET from v3, and POST to v4, but the URL is otherwise the same.